src/app/core/admin/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -0,0 +1,209 @@
+import { Component, inject, Input, OnInit } from "@angular/core";
+import { Location } from "@angular/common";
+import { DynamicComponentConfig } from "../../config/dynamic-components/dynamic-component-config.interface";
+import { DashboardConfig } from "../../dashboard/dashboard/dashboard.component";
+import { ConfigService } from "../../config/config.service";
+import { PREFIX_VIEW_CONFIG } from "../../config/dynamic-routing/view-config.interface";
+import { DynamicComponentDirective } from "../../config/dynamic-components/dynamic-component.directive";
+import {
+  CdkDragDrop,
+  DragDropModule,
+  moveItemInArray,
+} from "@angular/cdk/drag-drop";
+import { FaIconComponent } from "@fortawesome/angular-fontawesome";
+import { FormsModule } from "@angular/forms";
+import { MatButtonModule } from "@angular/material/button";
+import { MatCardModule } from "@angular/material/card";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { MatIconModule } from "@angular/material/icon";
+import { MatDialog } from "@angular/material/dialog";
+import { firstValueFrom } from "rxjs";
+import {
+  AdminWidgetDialogComponent,
+  AdminWidgetDialogData,
+} from "../admin-widget-dialog/admin-widget-dialog.component";
+import { ViewTitleComponent } from "../../common-components/view-title/view-title.component";
+import { WidgetComponentSelectComponent } from "../admin-entity-details/widget-component-select/widget-component-select.component";
+import { MatSnackBar } from "@angular/material/snack-bar";
+
+@Component({
+  selector: "app-admin-dashboard",
+  standalone: true,
+  imports: [
+    DynamicComponentDirective,
+    DragDropModule,
+    FaIconComponent,
+    FormsModule,
+    MatButtonModule,
+    MatCardModule,
+    MatTooltipModule,
+    MatIconModule,
+    ViewTitleComponent,
+  ],
+  templateUrl: "./admin-dashboard.component.html",
+  styleUrls: [
+    "./admin-dashboard.component.scss",
+    "../../dashboard/dashboard/dashboard.component.scss",
+    "../admin-entity/admin-entity-styles.scss",
+  ],
+})
+export class AdminDashboardComponent implements OnInit {
+  @Input() dashboardViewId: string;
+  @Input() isDisabled: boolean = false;
+
+  dashboardConfig: DashboardConfig;
+  private originalDashboardConfig: DashboardConfig;
+
+  private readonly configService = inject(ConfigService);
+  private readonly dialog = inject(MatDialog);
+  private readonly location = inject(Location);
+  private readonly snackBar = inject(MatSnackBar);
+
+  ngOnInit() {
+    this.loadDashboardViewConfig();
```

Comment:
if you move the code of the implementation for `loadDashboardViewConfig` immediately after this method, it is slightly easier to find & read for someone trying to understand the class